components/InternalLink.tsx

```diff
@@ -0,0 +1,52 @@
+import PropTypes from "helpers/PropTypes";
+import { Link } from "lbh-frontend-react";
+import querystring from "querystring";
+import React from "react";
+
+interface Url {
+  pathname: string;
+  query?: { [s: string]: string };
+}
+
+interface Props {
+  url: {
+    href: Url;
+    as: Url;
+  };
+}
+
+const combinePathAndQueryString = (url: Url): string => {
+  return url.query && Object.keys(url.query).length > 0
+    ? `${url.pathname}?${querystring.stringify(url.query)}`
+    : url.pathname;
+};
+
+const InternalLink: React.FunctionComponent<Props> = (props) => {
+  const { url } = props;
+  const { as } = url;
+
+  return (
+    <Link href={combinePathAndQueryString(as)}>
+      <a>Change</a>
+    </Link>
```

Comment:
```suggestion
    <Link href={combinePathAndQueryString(as)}>
      {children}
    </Link>
```

Where `children` in this case is `"Change"`. `Link` already includes the `<a>`.